OpenWeather/App/Favorites/FavoritesViewModel.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  OpenWeather template generated by OpenBytes on 16/03/2023.
+//
+// Created by Ahmed Shendy.
+//  FavoritesViewModel.swift
+//
+
+import Combine
+import ViewModel
+import Foundation
+import CoreLocation
+
+final class FavoritesViewModel: ViewModel<
+    FavoritesViewModel.Capabilities,
+    FavoritesViewModel.Input,
+    FavoritesViewModel.Content
+> {
+    struct Capabilities {
+        static var mock: Capabilities {
+            .init(
+                locationProviding: MockLocationProvider(),
+                weatherProviding: MockWeatherProvider(),
+                database: MockDatabaseService()
+            )
+        }
+
+        private let weatherProviding: WeatherProviding
+        private let locationProviding: LocationProviding
+        private let database: DatabaseService
+
+        init(
+            locationProviding: LocationProviding,
+            weatherProviding: WeatherProviding,
+            database: DatabaseService
+        ) {
+            self.locationProviding = locationProviding
+            self.weatherProviding = weatherProviding
+            self.database = database
+        }
+
+        func locationName(for location: CLLocation) async throws -> String {
+            try await locationProviding.locationName(for: location)
+        }
+
+        func weather(for location: CLLocation) async throws -> DeviceWeather {
+            try await weatherProviding.weather(for: location)
+        }
+
+        func getFavorites() async throws -> [DeviceLocation] {
+            try await database.fetchAllFavorites()
+        }
+
+        func getFavoritesPublisher() -> AnyPublisher<[DeviceLocation], Error> {
+            database.fetchAllFavoritesPublisher()
+        }
```

Comment:
This seems weird that we use Combine for this one only. Why is this? For any potential updates?